implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -493,4 +524,23 @@ private void storeMetadataLine(Metric.Builder metricBuilder, Map<String, String>
         }
     }
 
+    private String extractMetricKey(String metadataLine) {
```

Comment:
Can't `MetadataStep` do this for us instead of encoding it into a string and then trying to decode it again?
This also feels quite brittle. Btw should the index start at 1?

Reply:
> Btw should the index start at 1? 

Yes that's on purpose, the first index will always be `#`, I added a clarifying comment in [0d6d7f0](https://github.com/micrometer-metrics/micrometer/pull/4064/commits/0d6d7f059b7f8774f1230f42e9bc02c36bbcbbc5)